### PR TITLE
Improved error reporting when decorator manager applies a duplicate decorator

### LINKED ISSUE #711: Improve error reporting when decorator manager applies a duplicate decorator
# Feature Request 🛍️

If a decorator command set APPENDS two decorators with the same name the resulting model is invalid.

## Use Case

Improve error reporting.

## Possible Solution

Fail-fast, when the decorator is being added, with a clearer error message.

## Context

Apply this decorator command twice to the model.

```
      {
        "$class": "org.accordproject.decoratorcommands@0.2.0.Command",
        "type": "APPEND",
        "target": {
          "$class": "org.accordproject.decoratorcommands@0.2.0.CommandTarget",
          "namespace": "test@1.0.0",
          "declaration": "Driver",
          "property": "favoriteColor" // if you change this to 'foo' then validateCommands will throw an error
        },
        "decorator": {
          "$class": "concerto.metamodel@1.0.0.Decorator",
          "name": "Preference",
          "arguments": []
        }
      }
```

Currently the error is detected when the new model is created, rather than when it is decorated.

```
IllegalModelException: Duplicate decorator Preference Line 11 column 3, to line 12 column 1. 
    at Field.validate (/home/runner/AccordProjectConcerto-Decorator-Command-Set/node_modules/@accordproject/concerto-core/lib/introspect/decorated.js:130:23)
    at Field.validate (/home/runner/AccordProjectConcerto-Decorator-Command-Set/node_modules/@accordproject/concerto-core/lib/introspect/property.js:139:15)
    at ConceptDeclaration.validate (/home/runner/AccordProjectConcerto-Decorator-Command-Set/node_modules/@accordproject/concerto-core/lib/introspect/classdeclaration.js:283:23)
    at ModelFile.validate (/home/runner/AccordProjectConcerto-Decorator-Command-Set/node_modules/@accordproject/concerto-core/lib/introspect/modelfile.js:279:30)
    at ModelManager.validateModelFiles (/home/runner/AccordProjectConcerto-Decorator-Command-Set/node_modules/@accordproject/concerto-core/lib/basemodelmanager.js:397:33)
    at ModelManager.fromAst (/home/runner/AccordProjectConcerto-Decorator-Command-Set/node_modules/@accordproject/concerto-core/lib/basemodelmanager.js:754:14)
    at Function.decorateModels (/home/runner/AccordProjectConcerto-Decorator-Command-Set/node_modules/@accordproject/concerto-core/lib/decoratormanager.js:115:25)
    at Object.<anonymous> (/home/runner/AccordProjectConcerto-Decorator-Command-Set/index.js:109:44)
```

### PR DESCRIPTION
**Pull request by subhajit20**
# Closes #711 

- Improved error reporting when the decorator manager applies a duplicate decorator
- Added test for duplicate decorator

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

File: packages/concerto-core/lib/decoratormanager.js [hunk 1]
```diff
@@ -432,6 +433,14 @@ class DecoratorManager {
                     : (decorated.decorators = [newDecorator]);
             }
         } else if (type === 'APPEND') {
+            decorated.decorators.forEach((d)=>{
+                if(d === newDecorator){
```
Comment (mttrbrts) [inline packages/concerto-core/lib/decoratormanager.js:437]: We should match against the decorator name instead. Two decorators with the same name but different arguments should still be invalid.

**Review by mttrbrts — APPROVED** [review]
Thanks @subhajit20, please also add a unit test for your change.

File: packages/concerto-core/lib/decoratormanager.js (hunk 1 above)
Comment (subhajit20) [reply packages/concerto-core/lib/decoratormanager.js]: hey @mttrbrts done this and I am still confused about the test I have written... can you help out?

**Review by mttrbrts — APPROVED** [review]

**Comment by mttrbrts** [comment]
@subhajit20 There are some linting errors that are causing your build to fail. 

```
/home/runner/work/concerto/concerto/packages/concerto-core/lib/decoratormanager.js
Error:    24:39  error  Strings must use singlequote  quotes
Error:    24:76  error  Missing semicolon             semi
Error:   443:15  error  Missing semicolon             semi
```

**Comment by subhajit20** [comment]
> @subhajit20 There are some linting errors that are causing your build to fail.
> 
> ```
> /home/runner/work/concerto/concerto/packages/concerto-core/lib/decoratormanager.js
> Error:    24:39  error  Strings must use singlequote  quotes
> Error:    24:76  error  Missing semicolon             semi
> Error:   443:15  error  Missing semicolon             semi
> ```

Hey I have fixed those linting error.

**Comment by subhajit20** [comment]
I have added the test now.
Could you please review once you have time @mttrbrts

**Pull request closed without merging**